packages/skill-lesson/markdown/shiki-remote-plugin.ts

```diff
@@ -32,11 +33,39 @@ export interface ShikiRemotePluginOptions {
   theme?: string
 }
 
+type Highlighter = Awaited<ReturnType<typeof createShikiHighlighter>>
+
+let highlighter: Highlighter | undefined = undefined
+
 export function shikiRemotePlugin(opts: ShikiRemotePluginOptions): Transformer {
   return async (ast) => {
     await visitCodeNodes(ast, async (node) => {
       const code = node.value
 
+      if (!highlighter) {
+        highlighter = await createShikiHighlighter({
+          theme: opts.theme || 'dark-plus',
+        })
+      }
+
+      if (!node.meta?.includes('twoslash')) {
```

Comment:
We only need to call the remote service when necessary: when 'twoslash' is specified in the code fence. Otherwise, we can just run `renderCodeToHTML` which just acts like a normal, non-powerful syntax highlighter.